src/tools/fuzzing.h

```diff
@@ -2677,14 +2669,25 @@ class TranslateToFuzzReader {
 
   // special getters
   std::vector<Type> getSingleConcreteTypes() {
-    return items(
-      FeatureOptions<Type>()
-        .add(FeatureSet::MVP, Type::i32, Type::i64, Type::f32, Type::f64)
-        .add(FeatureSet::SIMD, Type::v128)
-        .add(FeatureSet::ReferenceTypes, Type::funcref, Type::externref)
-        .add(FeatureSet::ReferenceTypes | FeatureSet::ExceptionHandling,
-             Type::exnref)
-        .add(FeatureSet::ReferenceTypes | FeatureSet::Anyref, Type::anyref));
+    std::vector<Type> options{Type::i32, Type::i64, Type::f32, Type::f64};
+    if (wasm.features.hasSIMD()) {
+      options.push_back(Type::v128);
+    }
+    if (wasm.features.hasReferenceTypes()) {
+      addSingleConcreteReferenceTypes(options);
+    }
+    return options;
+  }
+
+  template<typename T> void addSingleConcreteReferenceTypes(T& options) {
+    options.push_back(Type::funcref);
+    options.push_back(Type::externref);
+    if (wasm.features.hasExceptionHandling()) {
+      options.push_back(Type::exnref);
+    }
+    if (wasm.features.hasAnyref() || wasm.features.hasGC()) {
+      options.push_back(Type::anyref);
+    }
```

Comment:
Other methods have the form of `get***Types` and return a vector, so maybe it'd be better to be consistent. Also we may not need `Concrete`, given that all reference types are concrete.

Reply:
This template is used in two places, one is directly above in `getSingleConcreteTypes`, where a `std::vector` is populated, and the other is in `makeRefIsNull` a little further up where a `SmallVector` is populated. Refactored this to deduplicate code adding reference types, so we only need to update one location in the future. More a helper to populate an existing vector of a varying type than an actual `get***Types` method. Perhaps `addSingleConcreteReferenceTypesTo` or `populateSingleReferenceTypes` to make it clearer?

Reply:
How about keeping `getSingleConcreteTypes` as is (there are only two lines of repetition, so I think it's fine), and creating `getReferenceTypes` and `getReferenceType` in the same format with other methods, and making `makeRefIsNull` use that? It will make `makeRefIsNull` even simpler. So it will be like

```cpp
std::vector<Type> getReferenceTypes() {
  return items(
    FeatureOptions<Type>()
      .add(FeatureSet::ReferenceTypes, Type::funcref, Type::externref)
      .add(FeatureSet::ReferenceTypes | FeatureSet::ExceptionHandling,
           Type::exnref)
      .add(FeatureSet::ReferenceTypes | FeatureSet::Anyref, Type::anyref));
}

Type getReferenceType() { return pick(getReferenceTypes()); }

Expression* makeRefIsNull(Type type) {
  assert(type == Type::i32);
  assert(wasm.features.hasReferenceTypes());
  return builder.makeRefIsNull(make(getReferenceType()));
}
```

Reply:
I expect more repetition once `eqref`, `i31ref` and the compound GC types are added. I don't feel strongly at this point, though, so going to change according to your suggestion.

Reply:
I see. We can change this later if necessary :)